assets/js/googlesitekit/datastore/user/authentication.test.js

```diff
@@ -0,0 +1,358 @@
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+		describe( 'fetchAuthentication', () => {
+			it( 'does not require any params', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).fetchAuthentication();
+				} ).not.toThrow();
+			} );
+		} );
+		describe( 'receiveAuthentication', () => {
+			it( 'requires the authentication param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).receiveAuthentication();
+				} ).toThrow( 'authentication is required.' );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getAuthentication', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				fetch
+					.doMockOnceIf( coreUserDataEndpointRegExp )
+					.mockResponseOnce(
+						JSON.stringify( coreUserDataExpectedResponse ),
+						{ status: 200 }
+					);
+
+				const initialConnection = registry.select( STORE_NAME ).getAuthentication();
+				// The authentication info will be its initial value while the authentication
+				// info is fetched.
+				expect( initialConnection ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getAuthentication() !== undefined
+					),
+				);
+
+				const authentication = registry.select( STORE_NAME ).getAuthentication();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( authentication ).toEqual( coreUserDataExpectedResponse );
+
+				const authenticationSelect = registry.select( STORE_NAME ).getAuthentication();
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( authenticationSelect ).toEqual( authentication );
+			} );
+
+			it( 'does not make a network request if data is already in state', async () => {
+				registry.dispatch( STORE_NAME ).receiveAuthentication( coreUserDataExpectedResponse );
+
+				const authentication = registry.select( STORE_NAME ).getAuthentication();
+
+				await subscribeUntil( registry, () => registry
+					.select( STORE_NAME )
+					.hasFinishedResolution( 'getAuthentication' )
+				);
+
+				expect( fetch ).not.toHaveBeenCalled();
+				expect( authentication ).toEqual( coreUserDataExpectedResponse );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf( coreUserDataEndpointRegExp )
+					.mockResponseOnce(
+						JSON.stringify( response ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				registry.select( STORE_NAME ).getAuthentication();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => store.getState().isFetchingAuthentication === false,
+				);
```

Comment:
Nitpick: we do this in a number of places but it isn't necessary to reference this in the store directly and I don't think there's really any place where we actually need to.

We should be using `hasFinishedResolution( 'getAuthentication' )` like above or using another selector.